src/components/e-table.vue

```diff
@@ -163,32 +437,136 @@
+
+        return count;
+      },
+    },
+    watch: {
+      /**
+       * Observes the current viewport and sets flag to show table sorting as a collapsible menu.
+       */
+      isMobile: {
+        immediate: true,
+        handler(): void {
+          this.showSortingOptions = !this.isMobile;
+        },
+      },
     },
-    // watch: {},
+
     // beforeCreate() {},
-    // created() {},
+    created() {
+      if (this.isMobile) {
+        window.addEventListener('resizeend', this.updateToggleButtonHeight);
+      } else {
+        window.removeEventListener('resizeend', this.updateToggleButtonHeight);
+      }
+    },
     // beforeMount() {},
     // mounted() {},
     // beforeUpdate() {},
     // updated() {},
     // activated() {},
     // deactivated() {},
-    // beforeUnmount() {},
+    beforeUnmount() {
+      window.removeEventListener('resizeend', this.updateToggleButtonHeight);
+    },
     // unmounted() {},
 
     methods: {
+      /**
+       * Returns the title of the actual table column.
+       */
+      columnTitle(column: IETableColumn): string | null {
+        switch (typeof column?.title) {
+          case 'string':
+            return column.title;
+
+          case 'function':
+            return column.title();
+
+          default:
+            return null;
+        }
+      },
+
+      /**
+       * Toggle the visibility for the sorting options.
+       */
+      toggleSortingOptions(): void {
+        if (this.isMobile) {
+          this.updateToggleButtonHeight();
+        }
+        this.showSortingOptions = !this.showSortingOptions;
+      },
+
+      /**
+       * Get the height of the sorting list toggle button.
+       */
+      updateToggleButtonHeight(): void {
+        this.toggleButtonHeight = this.toggleButton?.clientHeight || 0;
+      },
+
+      /**
+       * Returns a title for the row link, based on the type of the definition.
+       */
+      rowTitle(item: IETableItem): string | null {
+        const { rowLink } = this;
+
+        switch (typeof rowLink?.title) {
+          case 'string':
+            return rowLink.title;
+
+          case 'function':
+            return rowLink.title(item) || null;
+
+          default:
+            return null;
+        }
+      },
+
+      /**
+       * Checks if the given column should display the header label.
+       */
+      isHeaderLabelVisible(column: IETableColumn): boolean {
+        // Adding the support for functions was needed, to change visibility state dynamically (improved a11y).
+        return !!(typeof column.titleHidden === 'function' ? column.titleHidden() : column.titleHidden !== true);
+      },
+
+      /**
+       * Checks if the table is sorted by a given column.
+       *
+       * Since Vue3 leverages proxies for data properties for reactivity, we can't compare the objects directly.
+       */
+      isSortedBy(column: IETableColumn): boolean {
+        return JSON.stringify(this.sortBy) === JSON.stringify(column);
```

Comment:
todo: please use the `this.itemIdentifier` instead to get a value which can be compared.

Reply:
I have used the `key` property instead.